### PR TITLE
Focusable: Fix handling of `visibility: inherit`

### PR DESCRIPTION
**Pull request by scottgonzalez**
Ref #14596
Ref gh-1583

### CODE REVIEW

**Comment by jzaefferer** [comment]
Tests?

**Comment by jzaefferer** [comment]
Nevermind, saw your comment about IE8. Looks good to me.

**Comment by scottgonzalez** [comment]
I can add a test that explicitly assigned `visibility: inherit` which would cause all browsers to hit this code path.

**Comment by jzaefferer** [comment]
That seems useful.

File: ui/focusable.js [hunk 1]
```diff
@@ -42,9 +42,18 @@ $.ui.focusable = function( element, hasTabindex ) {
 		"a" === nodeName ?
 			element.href || hasTabindex :
 			hasTabindex ) &&
-		$( element ).is( ":visible" ) && $( element ).css( "visibility" ) === "visible";
+		$( element ).is( ":visible" ) && visible( $( element ) );
 };
 
+function visible( element ) {
+	var visibility = element.css( "visibility" );
+	while ( visibility === "inherit" ) {
```
Comment (westonruter) [inline ui/focusable.js]: Should this also check for `element.length !== 0` to avoid a possible infinite loop?

Comment (scottgonzalez) [inline ui/focusable.js]: There is no possible infinite loop here. If the element doesn't exist, it can't have a `visibility` of `"inherit"`.

Comment (westonruter) [inline ui/focusable.js]: Of course. :+1:

**Comment by scottgonzalez** [comment]
Tests added. It turns out that even with these tests, only IE 8 will hit the new code path. All other browsers resolve `inherit` to either `hidden` or `visible` when getting the computed value (which makes sense). As such, I've marked the `visible()` method as supporting IE 8, but not the tests. The test cases are still valid, and if the implementation were to ever change to checking `element.style` instead of the computed style, the tests would catch the failure (unless the tree walking logic still existed).

**Comment by jzaefferer** [comment]
Looks good. Somewhat related: https://github.com/medialize/ally.js/issues/41

**Pull request closed without merging**